torch/_inductor/runtime/triton_heuristics.py

```diff
@@ -481,12 +481,41 @@ def _precompile_config(self, cfg: Config, warm_cache_only: bool):
                 raise
             binary._init_handles()
 
+        """
+        https://github.com/pytorch/pytorch/issues/115344
+
+        self.fn.constexprs doesn't properly deal with None args, so when we filter out
+        an arg in UserDefinedTritonKernel.codegen, we need to filter it here as well.
+        We also don't want to modify self.fn.
+
+        We know that we removed something from the signature if:
+            1. It's in compile_meta["constants"]
+            2. It isn't a constant we already know about
+                Note: The value of interest has already been added to compile_meta['constants'],
+                    so we use self.fn.constexprs instead.
+            3. It isn't in the compile_meta signature
+        """
+        self.arg_names = self.fn.arg_names
```

Comment:
Why do we need this? It doesn't look to be changed anywhere, so should we keep using the `self.fn.arg_names` as before? If we really want a copy in `self.arg_names`, let's do `list(self.fn.arg_names)` to shallow-copy, so that we don't keep a reference (assuming `self.fn.arg_names` is a `list`, can lead to unexpected side effects down the road).

Reply:
it turned out that this is a bug, so we don't need this! (we can just use self.fn.arg_names)